cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -1052,6 +1069,7 @@ data IndexCacheEntry
 
 data NoIndexCacheEntry
     = CacheGPD GenericPackageDescription !BSS.ByteString
+    | NoIndexCachePreference Dependency
```

Comment:
I think this should be `NoIndexCachePreference [Dependency]` I.e. just single entry? I'd imagine it would make loading a lot faster.

Reply:
Changed to the suggested structure